packages/framework/client-logger/external-telemetry/src/container/containerTelemetry.ts

```diff
@@ -0,0 +1,170 @@
+	 * Name for the container telemetry event that is intended to be produced from the internal container "disconnected" system event
+	 *
+	 * @see {@link ContainerDisconnectedTelemetry}
+	 */
+	DISCONNECTED: "container.disconnected",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "closed" system event
+	 *
+	 * @see {@link ContainerClosedTelemetry}
+	 */
+	CLOSED: "container.closed",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "attaching" system event
+	 *
+	 * @see {@link ContainerAttachingTelemetry}
+	 */
+	ATTACHING: "container.attaching",
+	/**
+	 * Name for the container telemetry event that is intended to be produced from the internal container "attached" system event
+	 *
+	 * @see {@link ContainerAttachedTelemetry}
+	 */
+	ATTACHED: "container.attached",
+} as const;
+
+/**
+ * The type for all values within {@link ContainerTelemetryEventNames}
+ * @beta
+ */
+export type ContainerTelemetryEventName =
+	(typeof ContainerTelemetryEventNames)[keyof typeof ContainerTelemetryEventNames];
+
+/**
+ * The base interface for all Container telemetry
+ * @beta
+ */
+export interface IContainerTelemetry extends IExternalTelemetry {
+	/**
+	 * @inheritdoc IExternalTelemetry.eventName
+	 */
+	eventName: ContainerTelemetryEventName;
+	/**
+	 * The client side generated unique id of the container this telemetry came from.
+	 */
+	containerId: string;
+	/**
+	 * { @inheritdoc @fluidframework/container-definitions#IContainer."clientId" }
+	 */
+	clientId?: string;
+	/**
+	 * The id of the container this resolved url is for.
+	 * Will be undefined only when the container is not attached.
+	 */
+	documentId?: string;
+}
+
+/**
+ * The container "connected" telemetry event.
+ * It is produced from an internal Fluid container system event {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer} completes connecting to the Fluid service.
+ *
+ * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+ *
+ * @see
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * - {@link @fluidframework/container-definitions#IContainer.connect}
+ * @see {@link @fluidframework/container-definitions#IContainer.connectionState}
+ *
+ * @beta
+ */
+export interface ContainerConnectedTelemetry extends IContainerTelemetry {
+	eventName: "container.connected";
+}
+
+/**
+ * The container "disconnected" telemetry event. This telemetry is produced from an internal Fluid container system event
+ * {@link @fluidframework/container-definitions#IContainerEvents} which is emitted when the {@link @fluidframework/container-definitions#IContainer}
+ * becomes disconnected from the Fluid service.
+ *
+ * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
```

Comment:
It's a tricky subject to get into details of what disconnected actually means because it can mean different things at different layers. For now I'd skip this bit.
```suggestion
```

Reply:
Removed